datafusion/core/src/datasource/listing/url.rs

```diff
@@ -317,8 +315,27 @@ mod tests {
         let url = ListingTableUrl::parse("file:///foo/bar?").unwrap();
         assert_eq!(url.prefix.as_ref(), "foo/bar");
 
-        let url = ListingTableUrl::parse("file:///foo/😺").unwrap();
-        assert_eq!(url.prefix.as_ref(), "foo/%F0%9F%98%BA");
```

Comment:
This behaviour was pretty surprising IMO, you get back a path that is more percent encoded :exploding_head: